src/cycleclock.h

```diff
@@ -108,7 +120,9 @@ inline BENCHMARK_ALWAYS_INLINE int64_t Now() {
   asm volatile("mrs %0, cntvct_el0" : "=r"(virtual_timer_value));
   return virtual_timer_value;
 #elif defined(__ARM_ARCH)
-#if (__ARM_ARCH >= 6)  // V6 is the earliest arch that has a standard cyclecount
+  // V6 is the earliest arch that has a standard cyclecount
+  // Native Client validator doesn't allow MRC instructions.
+#if (__ARM_ARCH >= 6) && !defined(BENCHMARK_OS_NACL)
```

Comment:
Please restructure these ifdefs so that the most specialized implementations are at the top, and the generic one is the #else. If you have to add restrictions to the primary `#if` then it's probably better suited as an `#elif` or `#else`.